front/app/containers/Admin/projects/project/analysis/InputPreview/components/Comments/index.tsx

```diff
@@ -0,0 +1,66 @@
+import React, { useMemo, useEffect } from 'react';
+
+import { Box, Spinner, Title } from '@citizenlab/cl2-component-library';
+import { useInView } from 'react-intersection-observer';
+
+import useComments from 'api/comments/useComments';
+
+import { useIntl } from 'utils/cl-intl';
+
+import { useSelectedInputContext } from '../../../SelectedInputContext';
+
+import messages from './messages';
+import TopLevelComment from './TopLevelComment';
+
+const Comments = () => {
+  const { formatMessage } = useIntl();
+  const { selectedInputId } = useSelectedInputContext();
+  const { ref: loadMoreRef, inView } = useInView({
+    rootMargin: '200px',
+    threshold: 0,
+  });
+
+  const {
+    data: comments,
+    hasNextPage,
+    fetchNextPage,
+    isFetchingNextPage,
+  } = useComments({
+    ideaId: selectedInputId ?? undefined,
+    pageSize: 5,
+  });
+
+  const topLevelComments = useMemo(() => {
```

Comment:
The way this is set up is kind of silly. Basically, we fetch all comments (not just the top-level ones). Then we loop over these top-level comments, and for each comment we make another request for the child comments. But part of the child comments were already included in the original request for the top level comments, which is why we filter them out here...

So why did I do it like this? Because this is exactly how it works on the idea page. We can fix it here once we fix it there. We decided it was out of scope for this tandem.